README.md

```diff
@@ -247,6 +247,57 @@ Center(
 )
 ```
 
+## Concepts
+
+There are a few concepts that are useful to fully understand when using
+`flutter_portal`. That is especially true if you want to support custom use
+cases, which is easily possible with the abstract API provided.
+
+In the following, each of the abstract concepts you need to understand are
+explained on a high level. You will find them both in class names (e.g. the
+`Portal` widget or the `PortalTarget` widget as well as in parameter names).
+
+### Portal
+
+A portal (or the portal if you only have one) is the space used for doing all
+of the portal work. On a low level, this means that you have one widget that
+allows its subtree to place targets and followers that are connected.
+
+The portal also defines the area (rectangle bounds) that are available to any
+followers to be rendered onto the screen.
+
+In detail, you might wrap your whole `MaterialApp` in a single `Portal` widget,
+which would mean that you can use the whole area of your app to render followers
+attached to targets that are children of the `Portal` widget.
+
+### Target
+
+A target is any place within a portal that can be followed by a follower. This
```

Comment:
@creativecreatorormaybenot perhaps we can add 1-2 examples somewhere? For example an autocomplete dropdown and a textfield and a tooltip and an avatar. Both of these are examples of follower=>target.